### PR TITLE
Bridge support factory

### PR DESCRIPTION
**Pull request by nacho692**

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.ethereum.core.Repository;
+import org.ethereum.vm.LogInfo;
+import org.ethereum.vm.PrecompiledContracts;
+
+/**
+ * BridgeSupportFactory allows BridgeSupport instantiation.
+ */
+public class BridgeSupportFactory {
+
+    private Factory btcBlockStoreFactory;
```
Comment (collheesi) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java]: Always use `final` when possible

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        BridgeEventLogger eventLogger;
+        if (logs == null) {
+            eventLogger = null;
+        } else {
+            eventLogger = new BridgeEventLoggerImpl(bridgeConstants, logs);
+        }
+
+        return new BridgeSupport(bridgeConstants, provider, eventLogger, repository, executionBlock, btcContext,
+                federationSupport, btcBlockStoreFactory, null);
```
Comment (collheesi) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java]: Why is this parameter always null?

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package co.rsk.peg;
```
Comment (collheesi) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java]: Add license header

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            repository,
+            contractAddress,
+            bridgeConstants,
+            BridgeStorageConfiguration.fromBlockchainConfig(activations)
+        );
+
+        FederationSupport federationSupport = new FederationSupport(bridgeConstants, provider, executionBlock);
+
+        BridgeEventLogger eventLogger;
+        if (logs == null) {
```
Comment (collheesi) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java]: Where does this case come from?

Comment (nacho692) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java]: EthModule doesn't provide logs when creating the BridgeSupport instance. It might require a bigger refactor to avoid this.

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java [hunk 5]
```diff
@@ -97,15 +97,13 @@ public EthModule(
...
 
         BridgeStorageProvider provider = new BridgeStorageProvider(
                 repository,
                 PrecompiledContracts.BRIDGE_ADDR,
                 bridgeConstants,
-                new BridgeStorageConfiguration(
-                        activationConfig.isActive(ConsensusRule.RSKIP87, bestBlock.getNumber()),
-                        activationConfig.isActive(ConsensusRule.RSKIP123, bestBlock.getNumber())
-                )
+                BridgeStorageConfiguration.fromBlockchainConfig(activations)
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java]: Good duplication finding! maybe this hunk could go into a different commit
